File: Assets/zfoolua/common/PairLong.lua

local PairLong = {}

function PairLong:new()
    local obj = {
        key = 0, -- long
        value = 0 -- long
    }
    setmetatable(obj, self)
    self.__index = self
    return obj
end

function PairLong:protocolId()
    return 111
end

function PairLong:write(buffer, packet)
    if packet == nil then
        buffer:writeInt(0)
        return
    end
    buffer:writeInt(-1)
    buffer:writeLong(packet.key)
    buffer:writeLong(packet.value)
end

function PairLong:read(buffer)
    local length = buffer:readInt()
    if length == 0 then
        return nil
    end
    local beforeReadIndex = buffer:getReadOffset()
    local packet = PairLong:new()
    local result0 = buffer:readLong()
    packet.key = result0
    local result1 = buffer:readLong()
    packet.value = result1
    if length > 0 then
        buffer:setReadOffset(beforeReadIndex + length)
    end
    return packet
end

return PairLong

File: Assets/zfoocs/Common/PairLong.cs
using System;
using System.Collections.Generic;
namespace zfoocs
{
    
    public class PairLong
    {
        public long key;
        public long value;
    }

    public class PairLongRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 111;
        }
    
        public void Write(ByteBuffer buffer, object packet)
        {
            if (packet == null)
            {
                buffer.WriteInt(0);
                return;
            }
            PairLong message = (PairLong) packet;
            buffer.WriteInt(-1);
            buffer.WriteLong(message.key);
            buffer.WriteLong(message.value);
        }
    
        public object Read(ByteBuffer buffer)
        {
            int length = buffer.ReadInt();
            if (length == 0)
            {
                return null;
            }
            int beforeReadIndex = buffer.GetReadOffset();
            PairLong packet = new PairLong();
            long result0 = buffer.ReadLong();
            packet.key = result0;
            long result1 = buffer.ReadLong();
            packet.value = result1;
            if (length > 0)
            {
                buffer.SetReadOffset(beforeReadIndex + length);
            }
            return packet;
        }
    }
}